drupal/modules/avoindata-header/src/Plugin/Form/SearchForm.php

```diff
@@ -0,0 +1,86 @@
+<?php
+
+namespace Drupal\avoindata_header\Plugin\Form;
+
+use Drupal\Core\Form\FormBase;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Url;
+
+/**
+ * Adds search form to header element.
+ *
+ * Class SearchForm
+ *   Adds form.
+ *
+ * @package Drupal\avoindata_header\Plugin\Form
+ */
+class SearchForm extends FormBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getFormId() {
+    return 'avoindata_search_id';
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildForm(array $form, FormStateInterface $form_state) {
+    $form['language'] = \Drupal::languageManager()->getCurrentLanguage()->getId();
+
+    $form['searchfilter'] = [
+      '#type' => 'textfield',
+      '#default_value' => '1',
+      '#attributes' => ['class' => ['input-header-search-filter', 'hidden']],
+    ];
+
+    $form['#theme'] = ['avoindata_search'];
+
+    $form['#language'] = \Drupal::languageManager()->getCurrentLanguage()->getId();
+
+    $form['search'] = [
+      '#type' => 'textfield',
+      '#default_value' => '',
+    ];
+
+    $form['actions']['submit'] = [
+      '#type' => 'submit',
+      '#value' => $this->t('Submit'),
+      '#button_type' => 'primary',
+    ];
+
+    return $form;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function validateForm(array &$form, FormStateInterface $form_state) {
```

Comment:
Is this really needed? If not maybe it should be removed. If it is, maybe describe reason for it with comment

Reply:
Resolved in [92a479a](https://github.com/vrk-kpa/opendata/pull/1774/commits/92a479a6671ee9d1f0071c475d1135bf7a17fbd2)